### PR TITLE
Editorial: consider removing merchant preference language

### LINKED ISSUE #481: Consider removing merchant preference language
Section 3.3 item 10.

### PR DESCRIPTION
**Pull request by ianbjacobs**
- new goal about streamlined user experience
- neutralize pref language

closes https://github.com/w3c/browser-payment-api/issues/481

Based on WPWG teleconf discussion
http://www.w3.org/2017/04/06-wpwg-minutes

***
[Preview](https://s3.amazonaws.com/pr-preview/ianbjacobs/browser-payment-api/gh-pages.html) | [Diff](https://s3.amazonaws.com/pr-preview/w3c/browser-payment-api/42916cb...ianbjacobs:8f12a31.html)

### CODE REVIEW

**Comment by marcoscaceres** [comment]
@ianbjacobs, seems your version of tidy is different from the one we've been using. Make sure you have the latest from http://www.html-tidy.org/ or `brew install tidy-html5`. You should be on version 5.4.0.

**Review by marcoscaceres — CHANGES_REQUESTED** [review]
Can't view changes because of tidy.

**Comment by marcoscaceres** [comment]
Also, make sure you run:

```
tidy -config tidyconfig.txt -o index.html index.html
```

**Comment by ianbjacobs** [comment]
Hi @marcoscaceres,

I updated tidy (now 5.4.0) and ran it. Cheers!

Ian

**Comment by marcoscaceres** [comment]
@ianbjacobs, really sorry for the pain, but it's still not quite right. It seems you still have version 5.2 - and it didn't pick up our configuration file. Please make sure you run:

```BASH
tidy -config tidyconfig.txt -o index.html index.html
```

File: index.html [hunk 1]
```diff
@@ -1,6 +1,8 @@
 <!DOCTYPE html>
 <html>
   <head>
+    <meta name="generator" content=
+    "HTML Tidy for HTML5 for Mac OS X version 5.2.0">
```
Comment (marcoscaceres) [inline index.html]: Tidy mark should not be here.

**Review by marcoscaceres — CHANGES_REQUESTED** [review]

**Comment by ianbjacobs** [comment]
Hi @marcoscaceres,

Apparently tidy:no-mark does not cause a meta element to be removed if it is already there. It merely suppresses adding one if it is not there. I didn't realize this. I removed it by hand. (I confirm I have Tidy 5.4.)

**Comment by zkoch** [comment]
Barring issues with tidy, the text here LGTM

**Comment by mattsaxon** [comment]
With the revised spec text;
150 + Enable user agents to streamline the users's payment experience  
151 + by taking into account user preferences, merchant information,  
152 + security considerations, and other factors.  

Whilst it is clear to me how the other factor are exposed to the user-agent, there seems to me to be no way of the merchant expressing their preference.

I propose we make explicit that the ordering of the PMIs is that expression, but that we make no statement about the behaviour of the user-agent with regard to this ordering.

**Comment by mountainhippo** [comment]
That would be a good compromise for me. Appreciate we are walking a tightrope but we need to be able to provide a hint for merchant preference to maintain a capability that merchants have today

File: index.html [hunk 2]
```diff
@@ -147,6 +147,10 @@
           <li>Allow the user agent to act as intermediary between merchants,
           users, and <a>payment method</a> providers.
           </li>
+          <li>Enable user agents to streamline the users's payment experience
```
Comment (marcoscaceres) [inline index.html]: nit: s/users's/users'

File: index.html [hunk 3]
```diff
@@ -2702,12 +2701,12 @@
...
           one payment method identifier to try to determine what payment
           methods a <a>user agent</a> has installed. There may be legitimate
-          scenarios for calling repeatedly (for example, to control the order
-          of payment method selection). The fact that a successful match to a
-          payment method causes a user interface to be displayed mitigates the
-          disclosure risk. Implementations may also require a user action to
-          initiate a payment request or they may choose to rate limit the calls
-          to the API to prevent too many repeated calls.
+          scenarios for calling repeatedly (for example, to control the
+          sequence of payment method selection). The fact that a successful
```
Comment (marcoscaceres) [inline index.html]: "sequence" is problematic, because there is already WebIDL sequence defined.

**Review by marcoscaceres — CHANGES_REQUESTED** [review]

**Comment by ianbjacobs** [comment]
I don't think I closed this...not sure what happened.

**Comment by marcoscaceres** [comment]
The commit history vanished?

**Comment by ianbjacobs** [comment]
Did I screw something up (in GitHub)? :(

If so I am sorry. :(

Ian

**Comment by marcoscaceres** [comment]
I'm not sure - but no stress: just looks like the commits in the branch vanished. Let's see if we can get them back so not to lose the changes. Im on my phone right now - will send you some magic git commands once I get to work.

**Comment by ianbjacobs** [comment]
@mattsaxon, I believe you and Ken chatted about this. Based on that conversation are you satisfied with the closing of this issue with language as-is?

Ian

**Pull request closed without merging**